pkg/artifactory/resource_artifactory_backup.go

```diff
@@ -0,0 +1,251 @@
+				matchedBackup.SendMailOnError = iterBackup.SendMailOnError
+				for _, repo := range iterBackup.ExcludedRepositories.RepositoryRef {
+					matchedBackup.ExcludedRepositories = append(matchedBackup.ExcludedRepositories, repo)
+				}
+				break
+			}
+		}
+		var backupClass = ignoreHclPredicate("class", "rclass")
+		packer := universalPack(
+			allHclPredicate(
+				backupClass, schemaHasKey(backupSchema),
+			),
+		)
+		return diag.FromErr(packer(&matchedBackup, d))
+	}
+
+	var resourceBackupUpdate = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		unpackedBackup := unpackBackup(d)
+
+		/* EXPLANATION FOR BELOW CONSTRUCTION USAGE.
+		There is a difference in xml structure usage between GET and PATCH calls of API: /artifactory/api/system/configuration.
+		GET call structure has "backups -> backup -> Array of backup config blocks".
+		PATCH call structure has "backups -> Name/Key of backup that is being patched -> config block of the backup being patched".
+		Since the Name/Key is dynamic string, following nested map of string structs are constructed to match the usage of PATCH call.
+		*/
+		var constructBody = map[string]map[string]Backup{}
+		constructBody["backups"] = map[string]Backup{}
+		constructBody["backups"][unpackedBackup.Key] = unpackedBackup
+		content, err := yaml.Marshal(&constructBody)
+
+		if err != nil {
+			return diag.Errorf("failed to marshal backup config during Update")
+		}
+
+		err = sendConfigurationPatch(content, m)
+		if err != nil {
+			return diag.Errorf("failed to send PATCH request to Artifactory during Update")
+		}
+
+		// we should only have one backup config resource, using same id
+		d.SetId(unpackedBackup.Key)
+		return resourceBackupRead(ctx, d, m)
+	}
+
+	var resourceBackupDelete = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		backups := &Backups{}
+		rsrcBackup := unpackBackup(d)
+
+		response, err := m.(*resty.Client).R().SetResult(&backups).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+		if response.IsError() {
+			return diag.Errorf("got error response for API: /artifactory/api/system/configuration request during Read")
+		}
+
+		/* EXPLANATION FOR BELOW CONSTRUCTION USAGE.
+		There is a difference in xml structure usage between GET and PATCH calls of API: /artifactory/api/system/configuration.
+		GET call structure has "backups -> backup -> Array of backup config blocks".
+		PATCH call structure has "backups -> Name/Key of backup that is being patched -> config block of the backup being patched".
+		Since the Name/Key is dynamic string, following nested map of string structs are constructed to match the usage of PATCH call.
+		*/
+		var restoreBackups = map[string]map[string]Backup{}
+		restoreBackups["backups"] = map[string]Backup{}
+
+		for _, iterBackup := range backups.BackupArr {
```

Comment:
Please seperate the `find backup` from the `act on backup` code. In fact, I've seen this code 2x now - Time to define a function to simply find a backup